data-prepper-plugins/s3-source/src/main/java/org/opensearch/dataprepper/plugins/source/SqsWorker.java

```diff
@@ -128,58 +124,67 @@ private ReceiveMessageRequest createReceiveMessageRequest() {
                 .build();
     }
 
-    private Map<Message, List<S3EventNotification.S3EventNotificationRecord>> getS3MessageEventNotificationRecordMap(final List<Message> sqsMessages) {
-        return sqsMessages.stream().collect(Collectors.toMap(message -> message, this::convertS3EventMessages));
+    private Collection<ParsedMessage> getS3MessageEventNotificationRecords(final List<Message> sqsMessages) {
+        return sqsMessages.stream()
+                .map(this::convertS3EventMessages)
+                .collect(Collectors.toList());
     }
 
-    private List<S3EventNotification.S3EventNotificationRecord> convertS3EventMessages(final Message message) {
+    private ParsedMessage convertS3EventMessages(final Message message) {
         try {
             final S3EventNotification s3EventNotification = S3EventNotification.parseJson(message.body());
             if (s3EventNotification.getRecords() != null)
-                return s3EventNotification.getRecords();
+                return new ParsedMessage(message, s3EventNotification.getRecords());
             else {
                 LOG.debug("S3 notification message with ID:{} cannot be parsed into S3EventNotificationRecord.", message.messageId());
-                return Collections.emptyList();
+                return new ParsedMessage(message, true);
             }
 
         } catch (SdkClientException | JsonProcessingException e) {
-            LOG.error("Invalid JSON string in message body of {}", message.messageId(), e);
+            if (message.body().contains("s3:TestEvent") && message.body().contains("Amazon S3")) {
```

Comment:
QUES: why do we handle those events in exception handling instead of in the try block?

Reply:
The `s3:TestEvent` has a different JSON format from what we currently parse, so we can only do this by checking the string or parsing against a different JSON structure.

These test events are very rare. They are only created when S3 is configured to send messages to an SQS queue. Since they are rare, I took this approach so that the normal messages are handled first. It might yield a few benefits:
* Not incorrectly parsing a valid upload in the unlikely scenario that these strings are present.
* Avoid checking the string for something we probably know is not there.